File: internal/dto/platform.go
package dto

type ListedShopDTO struct {
	ShopID           uint64 `json:"shopId"`
	ShopName         string `json:"shopName"`
	PlatformTypeName string `json:"platformTypeName,omitempty"`
	PlatformTypeLogo string `json:"platformTypeLogo,omitempty"`
	SourceChannel    string `json:"sourceChannel,omitempty"`
}

type UpdateProductListingsRequest struct {
	ShopIDs []uint64 `json:"shopIds"`
}

type PlatformShopTypeDTO struct {
	ID         uint64 `json:"id,omitempty"`
	Code       string `json:"code" binding:"required"`
	Name       string `json:"name" binding:"required"`
	Logo       string `json:"logo"`
	Sort       int    `json:"sort"`
	Enabled    int8   `json:"enabled"`
	IsBuiltin  int8   `json:"isBuiltin,omitempty"`
	Remark     string `json:"remark"`
	ShopCount  int64  `json:"shopCount,omitempty"`
	CreateTime string `json:"createTime,omitempty"`
}

type PlatformShopQuery struct {
	Keyword        string `form:"keyword"`
	PlatformTypeID uint64 `form:"platformTypeId"`
	Status         *int8  `form:"status"`
	Page           int    `form:"page"`
	PageSize       int    `form:"pageSize"`
}

type PlatformShopDTO struct {
	ID                 uint64 `json:"id,omitempty"`
	PlatformTypeID     uint64 `json:"platformTypeId" binding:"required"`
	PlatformTypeName   string `json:"platformTypeName,omitempty"`
	PlatformTypeLogo   string `json:"platformTypeLogo,omitempty"`
	Name               string `json:"name" binding:"required"`
	SourceChannel      string `json:"sourceChannel,omitempty"`
	ShopCode           string `json:"shopCode"`
	ExternalShopID     string `json:"externalShopId"`
	Status             int8   `json:"status"`
	Remark             string `json:"remark"`
	Sort               int    `json:"sort"`
	ListedProductCount int64  `json:"listedProductCount,omitempty"`
	CreateTime         string `json:"createTime,omitempty"`
	UpdateTime         string `json:"updateTime,omitempty"`
}

File: web/src/types/platform.ts
export interface PlatformShopType {
  id: number
  code: string
  name: string
  logo?: string
  sort: number
  enabled: 0 | 1
  isBuiltin?: 0 | 1
  remark?: string
  shopCount?: number
  createTime?: string
}

export interface PlatformShop {
  id: number
  platformTypeId: number
  platformTypeName?: string
  platformTypeLogo?: string
  name: string
  sourceChannel?: string
  shopCode?: string
  externalShopId?: string
  status: 0 | 1
  remark?: string
  sort: number
  listedProductCount?: number
  createTime?: string
  updateTime?: string
}

export interface PlatformShopQuery {
  keyword?: string
  platformTypeId?: number
  status?: number
  page?: number
  pageSize?: number
}
